knowledge_repo/app/utils/knowledge_metadata.py

```diff
@@ -9,11 +9,11 @@
 class KnowledgeMetaExtension(Extension):
     """ Meta-Data extension for Python-Markdown. """
 
-    def extendMarkdown(self, md, md_globals):
+    def extendMarkdown(self, md):
         """ Add MetaPreprocessor to Markdown instance. """
-        md.preprocessors.add("knowledge_meta",
-                             KnowledgeMetaPreprocessor(md),
-                             ">normalize_whitespace")
+        md.preprocessors.register(KnowledgeMetaPreprocessor(md),
+                                  "knowledge_meta",
+                                  35)
```

Comment:
Can we restore this to the pre-patch version? If the priority of these other extensions changes, this will break; whereas if we keep the old `.add` syntax, we are insensitive to such changes. If we also add `md_globals=None` we can support the new and old version of `markdown` simultaneously.